### PR TITLE
Refactor boundary conditions (again)

### PR DESCRIPTION
**Pull request by psakievich**
Fix remaining issues with #998

### CODE REVIEW

**Review by rcknaus — APPROVED** [review]
Nice!

**Review by alanw0 — APPROVED** [review]
Looks like a nice cleanup.

File: src/BoundaryConditions.C [hunk 1]
```diff
@@ -7,133 +7,117 @@
...
-//--------------------------------------------------------------------------
-//-------- load -----------------------------------------------
-//--------------------------------------------------------------------------
+// factory method to create any supported bc
+std::unique_ptr<BoundaryCondition>
+BoundaryConditionCreator::load_single_bc_node(const YAML::Node& node)
+{
+  std::unique_ptr<BoundaryCondition> this_bc;
+  if (node["wall_boundary_condition"]) {
+    this_bc = std::move(register_bc<WallBoundaryConditionData>(node));
```
Comment (rcknaus) [inline src/BoundaryConditions.C:37]: Do you need the `move` here? register_bc returns an r-value, right?

**Review by jrood-nrel — APPROVED** [review]

File: src/BoundaryConditions.C (hunk 1 above)
Comment (psakievich) [reply src/BoundaryConditions.C:37]: No, right now it returns a `std::unique_ptr` but maybe it shouldn't

Comment (psakievich) [reply src/BoundaryConditions.C:37]: Relooking at this now and this is what I intended.  The `unique_ptr` gets passed through a couple of function calls, but ends up in the storage vector at the end.  A little messy still I guess, but functional.

Comment (psakievich) [reply src/BoundaryConditions.C:37]: Oh wait nvm. I think you are right. I'll update it

**Pull request merged**